OsmAnd/src/net/osmand/plus/onlinerouting/OnlineRoutingHelper.java

```diff
@@ -104,16 +105,41 @@ public OnlineRoutingResponse calculateRouteOnline(@Nullable String stringKey, @N
 	public OnlineRoutingResponse calculateRouteOnline(@NonNull OnlineRoutingEngine engine, @NonNull List<LatLon> path, @Nullable Float startBearing,
 													  boolean leftSideNavigation, boolean initialCalculation) throws IOException, JSONException {
 		String url = engine.getFullUrl(path, startBearing);
-		String content = makeRequest(url);
+		String method = engine.getHTTPMethod();
+		String body = engine.getRequestBody(path, startBearing);
+		Map<String, String> headers = engine.getRequestHeaders();
+		String content = makeRequest(url, method, body, headers);
 		return engine.parseResponse(content, app, leftSideNavigation, initialCalculation);
 	}
 
 	@NonNull
 	public String makeRequest(@NonNull String url) throws IOException {
+			return makeRequest(url, "GET", null, null);
+	}
+
+	@NonNull
+	public String makeRequest(@NonNull String url, @NonNull String method,
+							  @Nullable String body, @Nullable Map<String, String> headers)
+			throws IOException {
 		HttpURLConnection connection = NetworkUtils.getHttpURLConnection(url);
 		connection.setRequestProperty("User-Agent", Version.getFullVersion(app));
+		connection.setRequestMethod(method);
+		if (!method.equals("GET")) {
```

Comment:
Maybe also use `method.equals("POST")`, and if the headers block (see below) is moved outside combine to 
`method.equals("POST") && body != null`